modules/core/arrow-instances-core/src/main/kotlin/arrow/instances/char.kt

```diff
@@ -1,7 +1,31 @@
 package arrow.instances
 
 import arrow.typeclasses.Eq
+import arrow.typeclasses.Order
 
-object CharEqInstance : Eq<Char> {
-  override fun Char.eqv(b: Char): Boolean = this == b
+interface CharEqInstance : Eq<Char> {
+    override fun Char.eqv(b: Char): Boolean = this == b
 }
+
+interface CharOrderInstance : Order<Char> {
+    override fun Char.eqv(b: Char): Boolean = this == b
+
+    override fun Char.compare(b: Char): Int =
+            if (this < b) -1 else if (this > b) 1 else 0
+
+    override fun Char.lt(b: Char): Boolean = this < b
+
+    override fun Char.lte(b: Char): Boolean = this <= b
+
+    override fun Char.gt(b: Char): Boolean = this > b
+
+    override fun Char.gte(b: Char): Boolean = this >= b
+
+    override fun Char.neqv(b: Char): Boolean = this != b
+}
+
+fun Char.Companion.eq(): Eq<Char> =
+        lazyOf(object : CharEqInstance {}).value
```

Comment:
Why are these wrapped in `lazyOf` and then immediately extract the value?

Reply:
Unlike Scala, in Kotlin that `lazyOf` will be invoked on every function call. You need a separate field that's returned from the function.

Reply:
What do you mean by a separate field ? Can't you just return the object as is?
```kotlin
fun Char.Companion.eq(): Eq<Char> = object : CharEqInstance {}
```